# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.cpp
# Commit: 63d9972bab1f
# Full Hash: 63d9972bab1f267a940ec30c7cea7a4c41e54fa6
# Author: Alexandru Marc <amarc@mozila.com>
# Date: 2025-06-06 08:52:03
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing leaks @ mochitest.toml
#   
#   This reverts commit a97cb1d6c08bc38e8644bfc916298b847dbac3a1.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 801e25cb4bfd -r 63d9972bab1f dom/ipc/WindowGlobalParent.cpp
--- a/dom/ipc/WindowGlobalParent.cpp	Thu Jun 05 18:51:36 2025 +0000
+++ b/dom/ipc/WindowGlobalParent.cpp	Thu Jun 05 21:40:49 2025 +0300
@@ -28,7 +28,6 @@
 #include "mozilla/dom/MediaController.h"
 #include "mozilla/dom/NavigatorLogin.h"
 #include "mozilla/dom/WebAuthnTransactionParent.h"
-#include "mozilla/dom/WebIdentityParent.h"
 #include "mozilla/dom/WindowGlobalChild.h"
 #include "mozilla/dom/ChromeUtils.h"
 #include "mozilla/dom/UseCounterMetrics.h"
@@ -1448,6 +1447,76 @@
   return IPC_OK();
 }
 
+IPCResult WindowGlobalParent::RecvGetIdentityCredential(
+    IdentityCredentialRequestOptions&& aOptions,
+    const CredentialMediationRequirement& aMediationRequirement,
+    bool aHasUserActivation, const GetIdentityCredentialResolver& aResolver) {
+  IdentityCredential::GetCredentialInMainProcess(
+      DocumentPrincipal(), this->BrowsingContext(), std::move(aOptions),
+      aMediationRequirement, aHasUserActivation)
+      ->Then(
+          GetCurrentSerialEventTarget(), __func__,
+          [aResolver](const IPCIdentityCredential& aResult) {
+            return aResolver({Some(aResult), NS_OK});
+          },
+          [aResolver](nsresult aErr) {
+            aResolver({Maybe<IPCIdentityCredential>(Nothing()), aErr});
+          });
+  return IPC_OK();
+}
+
+IPCResult WindowGlobalParent::RecvStoreIdentityCredential(
+    const IPCIdentityCredential& aCredential,
+    const StoreIdentityCredentialResolver& aResolver) {
+  IdentityCredential::StoreInMainProcess(DocumentPrincipal(), aCredential)
+      ->Then(
+          GetCurrentSerialEventTarget(), __func__,
+          [aResolver](const bool& aResult) { aResolver(NS_OK); },
+          [aResolver](nsresult aErr) { aResolver(aErr); });
+  return IPC_OK();
+}
+
+IPCResult WindowGlobalParent::RecvDisconnectIdentityCredential(
+    const IdentityCredentialDisconnectOptions& aOptions,
+    const DisconnectIdentityCredentialResolver& aResolver) {
+  IdentityCredential::DisconnectInMainProcess(DocumentPrincipal(), aOptions)
+      ->Then(
+          GetCurrentSerialEventTarget(), __func__,
+          [aResolver](const bool& aResult) { aResolver(NS_OK); },
+          [aResolver](nsresult aErr) { aResolver(aErr); });
+  return IPC_OK();
+}
+
+IPCResult WindowGlobalParent::RecvPreventSilentAccess(
+    const PreventSilentAccessResolver& aResolver) {
+  nsIPrincipal* principal = DocumentPrincipal();
+  if (principal) {
+    nsCOMPtr<nsIPermissionManager> permissionManager =
+        components::PermissionManager::Service();
+    if (permissionManager) {
+      permissionManager->RemoveFromPrincipal(
+          principal, "credential-allow-silent-access"_ns);
+      aResolver(NS_OK);
+      return IPC_OK();
+    }
+  }
+
+  aResolver(NS_ERROR_NOT_AVAILABLE);
+  return IPC_OK();
+}
+
+mozilla::ipc::IPCResult WindowGlobalParent::RecvSetLoginStatus(
+    LoginStatus aStatus, const SetLoginStatusResolver& aResolver) {
+  nsIPrincipal* principal = DocumentPrincipal();
+  if (!principal) {
+    aResolver(NS_ERROR_DOM_NOT_ALLOWED_ERR);
+    return IPC_OK();
+  }
+  nsresult rv = NavigatorLogin::SetLoginStatus(principal, aStatus);
+  aResolver(rv);
+  return IPC_OK();
+}
+
 IPCResult WindowGlobalParent::RecvGetStorageAccessPermission(
     bool aIncludeIdentityCredential,
     GetStorageAccessPermissionResolver&& aResolve) {
@@ -1471,7 +1540,8 @@
 
   if (aIncludeIdentityCredential) {
     bool canCollect;
-    rv = identity::CanSilentlyCollect(topPrincipal, principal, &canCollect);
+    rv = IdentityCredential::CanSilentlyCollect(topPrincipal, principal,
+                                                &canCollect);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       aResolve(nsIPermissionManager::UNKNOWN_ACTION);
       return IPC_OK();
@@ -1754,11 +1824,6 @@
   return MakeAndAddRef<WebAuthnTransactionParent>();
 }
 
-already_AddRefed<PWebIdentityParent>
-WindowGlobalParent::AllocPWebIdentityParent() {
-  return MakeAndAddRef<WebIdentityParent>();
-}
-
 NS_IMPL_CYCLE_COLLECTION_CLASS(WindowGlobalParent)
 
 NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN_INHERITED(WindowGlobalParent,